### ISSUE TITLE
[FEATURE REQ] Allow user to specify the HTTP protocol to use when talking to Azure

### ISSUE THREAD

**Issue by ashbondu** [Azure/azure-libraries-for-java#1197]
**Is your feature request related to a problem? Please describe.**
All the Azure systems do not yet support HTTP/2 protocol fully. But Java11 supports the HTTP/2 by default and okhttp client switches to HTTP/2 protocol when running under Java11, causing issues with throttling.  

**Describe the solution you'd like**
Allow the caller to define the protocol as part of AzureConfigurable during Azure client creation.
Instead Azure SDK can restrict the usage of HTTP protocol to fully Azure supported HTTP protocols.

**Comment by ChenTanyi** [comment]
@ashbondu It is a feature in client-runtime, you could use a workaround as:
```java
            OkHttpClient.Builder httpClientBuilder = new OkHttpClient.Builder().protocols(Arrays.asList(Protocol.HTTP_1_1));

            RestClient client = new RestClient.Builder(httpClientBuilder, new Retrofit.Builder())
                    .withCredentials(credentials)
                    .build();

            Azure.authenticate(client, "<tenant id>").withSubscription("<subscription id>");
```
Default rest client configuration could be found in https://github.com/Azure/azure-libraries-for-java/blob/df7658ba/azure/src/main/java/com/microsoft/azure/management/Azure.java#L169

**Comment by ashbondu** [comment]
@ChenTanyi Thanks for the suggestion.
I am using this workaround today to restrict the protocol usage to HTTP/1.1 in our systems.

Just to give some background, we started seeing Azure throttling at low usage the system which was using the Azure client moved from Java8 to Java11 due to HTTP/1.1. Took us a bunch of days to narrow down that the reason is HTTP/2 

So, raised this more as an enhancement where Azure SDK can control the usage of the HTTP protocols more by either restricting the usage to fully supported protocols by Azure, so user is shielded from the pitfalls of system defaults not yet fully supported or by providing more control to the user via a builder function in AzureConfigurable builder.

**Comment by ChenTanyi** [comment]
@ashbondu Thanks for your feedback!

I also found some strange behavior when using OkHttp with Java 11. (Program isn't terminated after request is done.) Not sure whether it is the same or not.

Btw, I think my above solution can't be called workaround, it is what we expect currently.
Actually, the AzureConfigurable is used for common config, and the restClient providing all configs for OkHttp is recommended for some further usage like this situation.

Not sure whether this situation will become common or not.

**Comment by ChenTanyi** [comment]
@ashbondu Could you also tell me about which service you use in such Azure throttling? Or do you use multiple service? And how do you run with HTTP 2.0?

I think it could also to mark with service attention to help service debug if there is anything wrong with HTTP 2.0.

**Comment by ashbondu** [comment]
We use multiple services and many of the ARM calls were throttled at very low usage. 
Got throttled by GetVm, GetDisk, GetNic, GetResources etc.
Once we restricted the usage to HTTP/1.1 on the client, the throttling was not seen.
Have raised a Azure SR as well regarding this.

Regarding the usage of HTTP/2, as I mentioned earlier, Java11 supports HTTP/2 by default. So OkHttp started using HTTP/2 protocol by default to talk to Azure once we moved our systems from Java8 to Java11.
